### PR TITLE
Add error for user code sensors with too many entities targeted

### PR DESCRIPTION
**Pull request by OwenKephart**
## Summary & Motivation

Building off of the previous PR, now we emit a helpful error message rather than just letting users run into the RESOURCE_EXHAUSTED error.

### CODE REVIEW

**Comment by OwenKephart** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25215?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#25215** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25215?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#25214** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25214?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#25211** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25211?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#25210** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25210?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#25203** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25203?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#25204](https://github.com/dagster-io/dagster/pull/25204) <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25204?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#25190** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25190?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#25187** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25187?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#25186** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25186?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#25184** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25184?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#25171** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25171?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#25170** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25170?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @OwenKephart and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: python_modules/dagster/dagster/_core/definitions/automation_condition_sensor_definition.py [hunk 1]
```diff
@@ -46,7 +48,14 @@ def _evaluate(sensor_def: "AutomationConditionSensorDefinition", context: Sensor
         emit_backfills=sensor_def.emit_backfills,
         default_condition=sensor_def.default_condition,
         logger=context.log,
-    ).evaluate()
+    )
+    check.invariant(
```
Comment (alangenfeld) [inline python_modules/dagster/dagster/_core/definitions/automation_condition_sensor_definition.py:52]: prefer a DagsterError here

File: python_modules/dagster/dagster/_core/definitions/automation_condition_sensor_definition.py [hunk 2]
```diff
@@ -46,7 +48,14 @@ def _evaluate(sensor_def: "AutomationConditionSensorDefinition", context: Sensor
...
+        evaluation_context.total_keys <= MAX_ENTITIES,
+        f'AutomationConditionSensorDefintion "{sensor_def.name}" targets {evaluation_context.total_keys} '
+        f"assets or checks, which is more than the limit of {MAX_ENTITIES}. Either set `use_user_code_server` to `False`, "
+        "or split this sensor up into smaller selections.",
```
Comment (alangenfeld) [inline python_modules/dagster/dagster/_core/definitions/automation_condition_sensor_definition.py:56]: maybe more detail on what splitting in to smaller selections means exactly or how to do it

**Review by alangenfeld — APPROVED** [review]

**Comment by OwenKephart** [comment]
### Merge activity

* **Oct 22, 12:50 PM EDT**: A user started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/25215).

**Pull request merged**